core/state_processor.go

```diff
@@ -74,7 +74,11 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 		misc.ApplyDAOHardFork(statedb)
 	}
 	// Handle upgrade build-in system contract code
-	systemcontracts.UpgradeBuildInSystemContract(p.config, block.Number(), statedb)
+	lastBlock := p.bc.GetBlockByHash(block.ParentHash())
+	if lastBlock == nil {
```

Comment:
Just thinking about edge case here, there is no need to check if the block itself is nil similar to how we are checking its parent to be nil?

Reply:
the parent may be pruned.

Reply:
Yes but the block itself is always non-nil in this case right?